Dockerfile

```diff
@@ -2,29 +2,56 @@ FROM ubuntu:16.04
 MAINTAINER Haris Amin <aminharis7@gmail.com>
 
 # Install related packages and set LLVM 3.6 as the compiler
-RUN apt-get update && \
-    apt-get install -y make libc6-dev wget clang-3.6 vim curl libedit-dev python2.7 python2.7-dev libicu-dev rsync libxml2 git libcurl4-openssl-dev && \
-    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-3.6 100 && \
-    update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-3.6 100 && \
-    rm -rf /var/lib/apt/lists/*
+RUN apt-get -q update && \
+    apt-get -q install -y \
+    make \
+    libc6-dev \
+    clang-3.6 \
+    vim \
```

Comment:
Just tested in a bare `ubuntu:16.04`, and installing just `vim` is ~57.87 MB, which is space users can't ever reclaim if it's in the base image, whereas the following is pretty trivial:

```dockerfile
FROM swift
RUN apt-get update -qq && apt-get install -yqq vim && rm -rf /var/lib/apt/lists/*
```

Combine that with automated builds (https://docs.docker.com/docker-hub/builds/) and repository links (https://docs.docker.com/docker-hub/builds/#repository-links) and it's reasonably easy to have an up-to-date image built `FROM` this one with `vim` pre-installed.